client/my-sites/earn/memberships/test/coupons-list.jsx

```diff
@@ -0,0 +1,337 @@
+	const queryAddCouponButton = () => screen.queryByRole( 'button', { name: 'Add a new coupon' } );
+	const queryAddEditCouponDialog = () => screen.queryByRole( 'dialog', { name: 'Add a coupon' } );
+	const getCodeTextbox = () =>
+		screen.getByRole( 'textbox', { name: 'Enter a custom coupon code' } );
+	const getDiscountTypeSelect = () => screen.getByRole( 'combobox', { name: 'Discount type' } );
+	const getDiscountPercentageInput = () =>
+		screen.getByRole( 'textbox', { name: 'Discount percentage' } );
+	const queryDiscountPercentageInput = () =>
+		screen.queryByRole( 'textbox', { name: 'Discount percentage' } );
+	const queryDiscountValueInput = () => screen.queryByRole( 'textbox', { name: 'Discount value' } );
+	const getLimitDurationToggle = () => screen.getByRole( 'checkbox', { name: 'Duration' } );
+	const getLimitDurationSelection = () =>
+		screen.getByRole( 'combobox', { name: 'Duration selection' } );
+	const getLimitCouponsToggle = () =>
+		screen.getByRole( 'checkbox', { name: 'Limit coupon to specific emails' } );
+	const getLimitCouponsTextInput = () =>
+		screen.getByRole( 'textbox', { name: 'Limit coupon to specific emails text input' } );
+
+	beforeEach( () => {
+		modalRoot = document.createElement( 'div' );
+		modalRoot.setAttribute( 'id', 'wpcom' );
+		document.body.appendChild( modalRoot );
+		Modal.setAppElement( modalRoot );
+	} );
+
+	afterEach( () => {
+		unmountComponentAtNode( modalRoot );
+		document.body.removeChild( modalRoot );
+		modalRoot = null;
+	} );
+
+	test( 'should render nothing when feature is disabled', () => {
+		const render = ( el, options ) =>
+			renderWithProvider( el, {
+				...options,
+				initialState: {
+					...initialState,
+					memberships: {
+						settings: {
+							1: {
+								couponsAndGiftsEnabled: false,
+								connectedAccountId: 123,
+								connectedAccountMinimumCurrency: {
+									USD: 0.5,
+								},
+							},
+						},
+						couponList: {
+							items: {},
+						},
+						productList: {
+							items: {
+								1: [ testProduct1, testProduct2 ],
+							},
+						},
+					},
+				},
+				reducers: {
+					ui: uiReducer,
+					memberships: membershipsReducer,
+					siteSettings: siteSettingsReducer,
+				},
+			} );
+		render( <CouponsList /> );
+
+		const missingAddCouponButton = queryAddCouponButton();
+		const missingAddEditCouponDialog = queryAddEditCouponDialog();
+		const missingDiscountValueInput = queryDiscountValueInput();
+		const missingDiscountPercentageInput = queryDiscountPercentageInput();
+
+		expect( missingAddEditCouponDialog ).not.toBeInTheDocument();
+		expect( missingAddCouponButton ).not.toBeInTheDocument();
+		expect( missingDiscountPercentageInput ).not.toBeInTheDocument();
+		expect( missingDiscountValueInput ).not.toBeInTheDocument();
+	} );
+
+	test( 'should render with an empty list when no coupons are provided', () => {
+		const render = ( el, options ) =>
```

Comment:
Same as above: can we abstract this into a helper function outside of the test so it could be reused in multiple tests?